src/api/procedures/modifyComplianceRequirement.ts

```diff
@@ -0,0 +1,106 @@
+import { assertRequirementsNotTooComplex } from '~/api/procedures/utils';
+import { PolymeshError, Procedure, SecurityToken } from '~/internal';
+import { ErrorCode, InputCondition, TxTags } from '~/types';
+import { ProcedureAuthorization } from '~/types/internal';
+import { requirementToComplianceRequirement, stringToTicker } from '~/utils/conversion';
+import { conditionsAreEqual, hasSameElements } from '~/utils/internal';
+
+export type ModifyComplianceRequirementParams = {
+  /**
+   * ID of the Compliance Requirement
+   */
+  id: number;
+  /**
+   * array of conditions to replace the existing array of conditions for the requirement(identified by `id`).
+   *   Conditions within a requirement are *AND* between them. This means that in order
+   *   for a transfer to comply with this requirement, it must fulfill *ALL* conditions
+   */
+  conditions: InputCondition[];
+};
+
+/**
+ * @hidden
+ */
+export type Params = ModifyComplianceRequirementParams & {
+  ticker: string;
+};
+
+/**
+ * @hidden
+ */
+export async function prepareModifyComplianceRequirement(
+  this: Procedure<Params, SecurityToken>,
+  args: Params
+): Promise<SecurityToken> {
```

Comment:
I am not sure if SecurityToken is the right return value.

Stuff like `modifyCorporateActionsAgent` returns `Promise<void>`, while `modifyInstruction` returns an Instruction since that was what was being changed.

I am thinking we should either return ComplianceRequirements or void here.

(createCheckpoint has an example of using a resolver to get at post transaction values)

Reply:
`void` makes more sense to me. Returning the same data is of no use/ is just redundant IMO.